src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentEventHandlerLoweringPass.cs

```diff
@@ -222,5 +263,49 @@ private static IReadOnlyList<IntermediateToken> GetAttributeContent(TagHelperDir
                 return node.FindDescendantNodes<IntermediateToken>();
             }
         }
+
+        private IntermediateNode RewriteParameterUsage(IntermediateNode parent, TagHelperDirectiveAttributeParameterIntermediateNode node)
+        {
+            // Now rewrite the node to look like:
+            //
+            // builder.AddEventPreventDefaultAttribute(2, "onclick", true); // If minimized.
+            // or
+            // builder.AddEventPreventDefaultAttribute(2, "onclick", someBoolExpression); // If a bool expression is provided in the value.
+
+            string eventHandlerMethod;
+            if (node.BoundAttributeParameter.Name == "preventDefault")
+            {
+                eventHandlerMethod = ComponentsApi.RenderTreeBuilder.AddEventPreventDefaultAttribute;
+            }
+            else if (node.BoundAttributeParameter.Name == "stopPropagation")
+            {
+                eventHandlerMethod = ComponentsApi.RenderTreeBuilder.AddEventStopPropagationAttribute;
+            }
+            else
+            {
+                // Unsupported event handler attribute parameter. This can only happen if bound attribute descriptor
+                // is configured to expect a parameter other than 'preventDefault' and 'stopPropagation'.
+                return node;
+            }
+
+            var result = new ComponentAttributeIntermediateNode(node)
```

Comment:
This kinda seems like it would benefit from being a new node type. We'd have a node for "attribute" and a node for "attribute modifier". I realize it would cause a lot more plumbing, so I don't feel super strongly about it.

Reply:
Yeah I thought about it but decided against for that very reason. we can always add one if we need in the future. Just didn't feel like this warranted that much investment.